main.rb

```diff
@@ -0,0 +1,118 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/ModuleLength
+module Enumerable
+  def my_each
+    i = 0
+    while i < to_a.length
+      yield(to_a[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    my_each do |item|
+      yield(item, i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    result = []
+    each do |e|
+      result << e if yield(e)
+    end
+    result
+  end
+
+  def my_all?(data = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == false }
+      return true
+    elsif data.nil?
+      my_each { |n| return false if n.nil? || n == false }
+    elsif !data.nil? && (data.is_a? Class)
+      my_each { |n| return false if n.class != data }
+    elsif !data.nil? && (data.is_a? Regexp)
+      my_each { |n| return false unless data.match(n) }
+    else
+      my_each { |n| return false if n != data }
+    end
+    true
+  end
+
+  def my_any?(data = nil)
+    if block_given?
+      my_each { |item| return true if yield(item) }
+    elsif data.nil?
+      my_each { |n| return true unless n.nil? }
+    elsif data.is_a? Class
+      my_each { |n| return true if n.instance_of?(data) }
+    else
+      my_each { |n| return true if n == data }
+    end
+    false
+  end
```

Comment:
- [ ] when no block or argument is given, returns false if at least one of the elements in the collection is not true

- [ ] when a class is passed as an argument, returns true if all of the elements in the collection are members of the Numeric class

- [ ] when a Regex is passed as an argument, returns true if any of the collection matches the Regex